### DISCUSSION TITLE
Quasar Dark Mode flickers white in the first milliseconds of a page load

### DISCUSSION THREAD

**Discussion by tctfox** [quasarframework/quasar#15233] [General - Components / Directives / etc]
Hei there

I am building an app where I can toggle the dark mode. I only just implemented this one feature. Now my problem is, that the dark mode only kicks in a fracture of a second to late and that results in a short white flicker. This is extremely bad for the eye and makes the user experience quite bad. 

Right now I save the user preference in a store. Then in the app component I have a onBeforeMount() hook that gets the store value and sets the dark mode. On change of the store value I switch the dark mode (that part works really well). Is there a way to fix this? I think the main issue is that the rendering doesnt wait on the app onBeforeMount() hook,

What could I do to remove this white flickering? Any help is appreciated.

App.vue file
```javascript
<template>
  <router-view />
</template>

<script setup lang="ts">

import {onBeforeMount} from 'vue';
import {useCounterStore} from 'stores/example-store';
import {useQuasar} from 'quasar';

const $q = useQuasar()
const store = useCounterStore()

onBeforeMount(() => {
  $q.dark.set(store.userPrefDarkMode)
})

</script>
```
my store
```javascript
import { defineStore } from 'pinia';
import { ref, watch } from 'vue';
import { useQuasar } from 'quasar';

export const useCounterStore = defineStore('counter', () => {
  const userPrefDarkMode = ref(true)
  const $q = useQuasar();

  if (localStorage.getItem('userPrefDarkMode')){
    userPrefDarkMode.value = JSON.parse(<string>localStorage.getItem('userPrefDarkMode'));
  }

  watch(
    userPrefDarkMode, (darkVal) => {
      localStorage.setItem('userPrefDarkMode', JSON.stringify(darkVal))
      $q.dark.set(darkVal)

    },
    {deep: true}
  );

  return { userPrefDarkMode }
})```

**Comment by suleimanas** [comment]
browser render html then css and js ...
the only thing fixed it for me is to have initial background color
in `index.template.html` or `index.html`
`<html style="background: #000">`
or
`<body style="background: #000">`
